llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -4651,6 +4651,15 @@ static SDValue lowerVECTOR_SHUFFLE(SDValue Op, SelectionDAG &DAG,
     IndexVT = IndexVT.changeVectorElementType(MVT::i16);
   }
 
+  // If the mask allows, we can do all the index computation in 16 bits.  This
+  // requires less work and less register pressure at high LMUL, and creates
+  // smaller constants which may be cheaper to materialize.
+  if (IndexVT.getScalarType().bitsGT(MVT::i16) && isUInt<16>(NumElts * 2) &&
```

Comment:
Why `NumElts * 2`? Shouldn't it be `NumElts - 1`?

Reply:
You're correct, and I will adjust.  In practice, I'm not too worried about vectors with 32k elements.  :)

Reply:
Agreed, I don't think MVT goes that high yet.